### DISCUSSION TITLE
关于合并 SJTUTeX 和 SJTUThesis 的提案

### DISCUSSION THREAD

**Discussion by skyzh** [sjtug/SJTUThesis#630] [灌水区]
SJTUThesis texmf 文件夹下的内容是由 SJTUTeX 项目的 `sjtuthesis.dtx` 生成的，现在开发同时在 SJTUThesis 和 SJTUTeX 项目下进行，似乎有些麻烦。因此这个提案的内容就是讨论如何将两个项目合并。

* 把 SJTUTeX 的 `sjtuthesis.dtx` 文件放到 SJTUThesis 的根目录
* 开发者发 PR 时，需要保证 texmf 下的文件是最新的。我们加上一个 CI 来检查 texmf 和 dtx 是否一致。
* 在 release 和 CI build 的时候，也打包一份 SJTUTeX 的帮助文档。

因为 dtx 文件不是很大，一般也没有直接修改的需求，所以应该不会影响到用户的使用流程。我们可以在现在的文档中加上修改 SJTUThesis / SJTUTeX 模版的方法（先改 dtx，然后生成 texmf）。

cc @AlexaraWu 感觉上这样开发可能会方便一点（但我其实也不太确定.png
可以征询一下大家的意见

**Comment by AlexaraWu** [comment]
持保留意见。

FYI: #462

**Comment by skyzh** [comment]
原来以前拆过（逃

那就暂时搁置吧.png

**Comment by skyzh** [comment]
SJTUBeamer 这边最近在试验单 repo 同时存模版和示例文档 https://github.com/sjtug/SJTUBeamer, 目测效果挺好。开发者直接用 l3build 一键生成所有需要的文档（CTAN zip, 示例文档），并永远保持仓库中的 sty 文件最新；用户直接用 latexmk 编译 tex 文档。感觉 SJTUThesis 也可以考虑这么操作（

**Comment by AlexaraWu** [comment]
可以试试
